src/pages/AppList.svelte

```diff
@@ -14,15 +14,34 @@
   function filterApps(filterText) {
     filteredApps = apps.filter((appItem) => appItem.name.includes(filterText));
   }
+  let showDeprecated = true;
 </script>
 
 <h2>Applications</h2>
-<FilterInput onChangeText={filterApps} />
-{#if apps}
+
+<label>
+  <input type="checkbox" bind:checked={showDeprecated} />
+  Show deprecated applications
+</label>
+
+{#if showDeprecated}
+  <FilterInput onChangeText={filterApps} />
+  {#if apps}
+    {#each filteredApps as app}
+      <p>
+        <a href="/apps/{app.name}">{app.name}</a>
+        {#if app.description}<i>{app.description}</i>{/if}
+      </p>
+    {/each}
+  {/if}
+{:else if apps}
+  <FilterInput onChangeText={filterApps} />
   {#each filteredApps as app}
-    <p>
-      <a href="/apps/{app.name}">{app.name}</a>
-      {#if app.description}<i>{app.description}</i>{/if}
-    </p>
+    {#if !app.deprecated}
```

Comment:
You can skip the if/else (and the duplicated logic for display which violates the [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) principle) by doing the following on display:

```suggestion
    {#if showDeprecated || !app.deprecated}
```

Alternately, we could probably filter the applications in the filterApps function above. I'd accept either solution for now.

Reply:
Edited as per your suggestion @wlach . :smile:

Reply:
Hi @mhmohona -- it looks like the logic is still duplicated unfortunately. To be clear, you should only *one* block that iterates through the app list and no higher level if/else (i.e. the logic should be substantially the same as before, just with that extra piece of logic that I showed above).

Reply:
Sorry sorry @wlach, at that time I was confused. :sweat_smile: 
I hope now I could do what you asked.

I did not know about DRY principle before. I really appriciate the way you are helping to learn, to improve.